modules/pico_dev_ppp.c

```diff
@@ -341,7 +345,7 @@ static int ppp_serial_send_escape(struct pico_device_ppp *ppp, void *buf, int le
     /* Start byte. */
     out_buf[0] = in_buf[0];
     for(i = 1, j = 1; i < (len - 1); i++) {
-        if (((in_buf[i] + 1u) >> 1) == 0x3Fu) {
+        if (should_escape(in_buf[i])) {
```

Comment:
This commit is breaking the unit tests. Did you forgot to shift it by one or could this be a mis-interpretation of you @danielinux ?